google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITBuiltInMetricsTest.java

```diff
@@ -0,0 +1,122 @@
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner.it;
+
+import static com.google.common.truth.Truth.assertWithMessage;
+
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.monitoring.v3.MetricServiceClient;
+import com.google.cloud.spanner.Database;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.IntegrationTestEnv;
+import com.google.cloud.spanner.ParallelIntegrationTest;
+import com.google.cloud.spanner.Statement;
+import com.google.common.base.Stopwatch;
+import com.google.monitoring.v3.ListTimeSeriesRequest;
+import com.google.monitoring.v3.ListTimeSeriesResponse;
+import com.google.monitoring.v3.ProjectName;
+import com.google.monitoring.v3.TimeInterval;
+import com.google.protobuf.util.Timestamps;
+import com.google.spanner.admin.database.v1.UpdateDatabaseDdlMetadata;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.threeten.bp.Duration;
+import org.threeten.bp.Instant;
+
+@Category(ParallelIntegrationTest.class)
+@RunWith(JUnit4.class)
+@Ignore("Built-in Metrics are not GA'ed yet. Enable this test once the metrics are released")
+public class ITBuiltInMetricsTest {
+
+  private static Database db;
+  @ClassRule public static IntegrationTestEnv env = new IntegrationTestEnv();
+
+  private static DatabaseClient client;
+
+  private static MetricServiceClient metricClient;
+
+  @BeforeClass
+  public static void setUp() throws IOException {
+    metricClient = MetricServiceClient.create();
+    env.getTestHelper().getOptions().toBuilder().setEnableBuiltInMetrics(true);
+    db = env.getTestHelper().createTestDatabase();
+    client = env.getTestHelper().getDatabaseClient(db);
+  }
+
+  @Test
+  public void testBuiltinMetricsWithDefaultOTEL() throws Exception {
+    // This stopwatch is used for to limit fetching of metric data in verifyMetrics
+    Stopwatch metricsPollingStopwatch = Stopwatch.createStarted();
+    Instant start = Instant.now().minus(Duration.ofMinutes(2));
+    Instant end = Instant.now().plus(Duration.ofMinutes(3));
+    ProjectName name = ProjectName.of(env.getTestHelper().getOptions().getProjectId());
+
+    TimeInterval interval =
+        TimeInterval.newBuilder()
+            .setStartTime(Timestamps.fromMillis(start.toEpochMilli()))
+            .setEndTime(Timestamps.fromMillis(end.toEpochMilli()))
+            .build();
+    String ddl =
+        "CREATE TABLE FOO ("
+            + "  K    STRING(MAX) NOT NULL,"
+            + "  V    INT64,"
+            + ") PRIMARY KEY (K)";
+    OperationFuture<Void, UpdateDatabaseDdlMetadata> op =
+        db.updateDdl(Collections.singletonList(ddl), null);
+    op.get();
```

Comment:
We don't seem to use this table. Can we skip the creation of it?

Reply:
removed. I was in the assumption that we need to have a table to run Select 1 query.